src/Data/Lens/Types.purs

```diff
@@ -35,27 +35,39 @@ type Optic' p s a = Optic p s s a a
 type Iso s t a b = forall p. Profunctor p => Optic p s t a b
 type Iso' s a = Iso s s a a
 
+-- | An isomorphism defined in terms of `Exchange` to be safe from
+-- | impredicativity issues in the type checker. See the `docs/` folder for a
+-- | more detailed explanation.
 type AnIso s t a b = Optic (Exchange a b) s t a b
 type AnIso' s a = AnIso s s a a
 
 -- | A lens.
 type Lens s t a b = forall p. Strong p => Optic p s t a b
 type Lens' s a = Lens s s a a
 
+-- | A lens defined in terms of `Shop` to be safe from impredicativity issues in
```

Comment:
Duplication here

Reply:
I also don't know if "impredicativity issues in the type checker" is the best phrasing. The only impredicativity issue I know of is that we have no support for it whatsoever. I would say something like "to work around impredicative polymorphism".